sdk/inc/azure/core/az_span.h

```diff
@@ -154,7 +154,7 @@ AZ_NODISCARD az_span az_span_init(uint8_t* ptr, int32_t size);
  * @return An #az_span over the byte buffer where the size is set to the string's length not
  * including the \0 terminator.
  */
-AZ_NODISCARD az_span az_span_from_str(char* str);
+AZ_NODISCARD az_span az_span_from_str(char const* str);
```

Comment:
Regarding CI failure on linux and other platforms: https://dev.azure.com/azure-sdk/public/_build/results?buildId=444448&view=logs&j=1b9af15f-e962-500e-484c-e22016023d07&t=9b41a56d-3bc7-5f69-8428-d149c928186f

In the implementation of `az_span_from_str`, we end up casting away the const which results in an error:
```text
/home/vsts/work/1/s/sdk/src/azure/core/az_span.c: In function ‘az_span_from_str’:
/home/vsts/work/1/s/sdk/src/azure/core/az_span.c:52:23: error: cast discards ‘const’ qualifier from pointer target type [-Werror=cast-qual]
   return az_span_init((uint8_t*)str, length);
```

Reply:
I'm cancelling this PR after digging further into the proposed design.  Casting away the const is not the right thing to do.  If we want const on certain APIs we need to revisit the overall span design